src/components/TransactionSender/TransactionSender.ts

```diff
@@ -22,126 +18,119 @@ import {
   const signedTransactions = useSelector(signedTransactionsSelector);
 
   const sendingRef = useRef(false);
-
-  const dispatch = useDispatch();
+  const sentSessionIds = useRef<string[]>([]);
 
   const clearSignInfo = () => {
     dispatch(clearAllTransactionsToSign());
     sendingRef.current = false;
   };
-  async function handleSendTransactions() {
+
+  const handleSendTransactions = useCallback(async () => {
     const sessionIds = Object.keys(signedTransactions);
+
     for (const sessionId of sessionIds) {
-      const sessionInformation = signedTransactions?.[sessionId];
+      const session = signedTransactions[sessionId];
       const skipSending =
-        sessionInformation?.customTransactionInformation?.signWithoutSending;
+        session?.customTransactionInformation?.signWithoutSending;
 
-      if (!sessionId || skipSending) {
-        optionalRedirect(sessionInformation);
+      if (!session || !sessionId || skipSending) {
+        optionalRedirect(session);
         continue;
       }
 
-      try {
-        const isSessionIdSigned =
-          signedTransactions[sessionId].status ===
-          TransactionBatchStatusesEnum.signed;
-        const shouldSendCurrentSession =
-          isSessionIdSigned && !sendingRef.current;
-        if (!shouldSendCurrentSession) {
-          continue;
-        }
-        const { transactions } = signedTransactions[sessionId];
+      if (sentSessionIds.current.includes(sessionId)) {
+        continue;
+      }
 
-        if (!transactions) {
-          continue;
+      const { transactions } = session;
+      if (!transactions) {
+        continue;
+      }
+
+      const isSessionIdSigned =
+        session.status === TransactionBatchStatusesEnum.signed;
+      const shouldSendCurrentSession = isSessionIdSigned && !sendingRef.current;
+
+      if (!shouldSendCurrentSession) {
+        continue;
+      }
+
+      sendingRef.current = true;
+      sentSessionIds.current.push(sessionId);
+
+      const responseHashes = await invokeSendTransactions({
+        session,
+        sessionId,
+        address,
+        clearSignInfo,
+        sendSignedTransactionsAsync,
+        sendSignedBatchTransactionsAsync
+      });
+
+      const newStatus = TransactionServerStatusesEnum.pending;
+      const newTransactions = transactions.map((transaction) => {
+        if (responseHashes?.includes(transaction.hash)) {
+          return { ...transaction, status: newStatus };
         }
-        sendingRef.current = true;
-        const transactionsToSend = transactions.map((tx) => newTransaction(tx));
-        const responseHashes = await sendSignedTransactionsAsync(
-          transactionsToSend
-        );
-
-        const newStatus = TransactionServerStatusesEnum.pending;
-        const newTransactions = transactions.map((transaction) => {
-          if (responseHashes.includes(transaction.hash)) {
-            return { ...transaction, status: newStatus };
```

Comment:
I think this was correct - maybe bring back?

Reply:
This is just refactored, not removed. The functionality is still there